akka-actor/src/main/scala/akka/actor/ActorCell.scala

```diff
@@ -684,25 +688,15 @@ private[akka] class ActorCell(
     case _                               =>
   }
 
-  final protected def clearActorCellFields(cell: ActorCell): Unit = {
-    cell.unstashAll()
-    if (!Reflect.lookupAndSetField(classOf[ActorCell], cell, "props", ActorCell.terminatedProps))
-      throw new IllegalArgumentException("ActorCell has no props field")
-  }
-
-  final protected def clearActorFields(actorInstance: Actor, recreate: Boolean): Unit = {
-    setActorFields(actorInstance, context = null, self = if (recreate) self else system.deadLetters)
+  final protected def clearActorFields(): Unit = {
     currentMessage = null
     behaviorStack = emptyBehaviorStack
   }
-
-  final protected def setActorFields(actorInstance: Actor, context: ActorContext, self: ActorRef): Unit =
-    if (actorInstance ne null) {
-      if (!Reflect.lookupAndSetField(actorInstance.getClass, actorInstance, "context", context)
-          || !Reflect.lookupAndSetField(actorInstance.getClass, actorInstance, "self", self))
-        throw IllegalActorStateException(
-          s"${actorInstance.getClass} is not an Actor class. It doesn't extend the 'Actor' trait")
-    }
+  protected def clearFieldsForTermination(): Unit = {
+    unstashAll()
+    _props = ActorCell.terminatedProps
+    actor = null
```

Comment:
isn't this `def actor`? how is it possible to set a def to null?
should it be `_actor = null`?

Reply:
I was thinking, how is that even compiling...

here is the reason:
```scala
protected def actor_=(a: Actor): Unit = _actor = a
```

Reply:
I often forget that Scala has syntax for that.

Reply:
ah, I didn't see that.
ok, doesn't have to change (but I think it's more clear to use explicit `_actor = null`

Reply:
I'll simplify it.